routemaster/state_machine/actions.py

```diff
@@ -104,16 +106,16 @@ def process_retries(
         )
 
     for label_name, metadata in relevant_labels.items():
-        label = LabelRef(name=label_name, state_machine=state_machine.name)
-        could_progress = False
+        with process_one_job():
+            label = LabelRef(name=label_name, state_machine=state_machine.name)
+            could_progress = False
 
-        with app.db.begin() as conn:
-            try:
+            with app.db.begin() as conn:
                 lock_label(label, conn)
                 current_state = get_current_state(label, state_machine, conn)
 
                 if current_state != action:
-                    continue
+                    return
```

Comment:
Why did this need to change from `continue` to `return`? Surely we don't want to stop processing for all the pending users just because one of them is in the wrong state?

Reply:
Good catch, leftover from refactoring. d77e59cf11f111a054672694a607276ba0f944bc